mne/utils/numerics.py

```diff
@@ -759,6 +759,12 @@ def object_diff(a, b, pre=''):
             for ii, (xx1, xx2) in enumerate(zip(a, b)):
                 out += object_diff(xx1, xx2, pre + '[%s]' % ii)
     elif isinstance(a, (str, int, float, bytes, np.generic)):
+        # catch non-array np.nan floats
+        if isinstance(a, float):
+            if np.isnan(a):
+                if not _array_equal_nan(a, b):
+                    out += pre + ' value mismatch (%s, %s)\n' % (a, b)
+                return out
```

Comment:
I don't like this short-circuiting logic -- now we have to know that there is another `return` path. Instead how about:
```suggestion
    elif isinstance(a, float):
        if not _array_equal_nan(a, b):
            out += pre + ' value mismatch (%s, %s)\n' % (a, b)
    elif isinstance(a, (str, int, bytes, np.generic)):
```
Basically if it's `float` we directly triage to a function that properly checks for equality. If it's any of the other types we can use the builtin `==`.

Reply:
yes, that looks a lot better, thanks!